src/mutation.rs

```diff
@@ -0,0 +1,225 @@
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(desc = "Uuid of order to place")] id: Uuid,
+    ) -> Result<Order> {
+        let db_client = ctx.data_unchecked::<Database>();
+        let collection: Collection<Order> = db_client.collection::<Order>("orders");
+        let order = query_order(&collection, id).await?;
+        authenticate_user(&ctx, order.user._id)?;
+        set_status_placed(&collection, id).await?;
+        query_order(&collection, id).await
+    }
+}
+
+/// Extracts UUID from Bson.
+///
+/// Adding a order returns a UUID in a Bson document. This function helps to extract the UUID.
+fn uuid_from_bson(bson: Bson) -> Result<Uuid> {
+    match bson {
+        Bson::Binary(id) => Ok(id.to_uuid()?),
+        _ => {
+            let message = format!(
+                "Returned id: `{}` needs to be a Binary in order to be parsed as a Uuid",
+                bson
+            );
+            Err(Error::new(message))
+        }
+    }
+}
+
+/// Sets the status of an order to `OrderStatus::Placed`.
+///
+/// * `collection` - MongoDB collection to update.
+/// * `input` - `UpdateOrderInput`.
+async fn set_status_placed(collection: &Collection<Order>, id: Uuid) -> Result<()> {
+    let current_timestamp = DateTime::now();
+    let result = collection
+        .update_one(
+            doc! {"_id": id },
+            doc! {"$set": {"order_status": OrderStatus::Placed, "placed_at": current_timestamp}},
+            None,
+        )
+        .await;
+    if let Err(_) = result {
+        let message = format!("Placing order of id: `{}` failed in MongoDB.", id);
+        return Err(Error::new(message));
+    }
+    Ok(())
+}
+
+/// Checks if foreign types exist (MongoDB database populated with events).
+async fn validate_input(db_client: &Database, input: &CreateOrderInput) -> Result<()> {
+    let user_collection: mongodb::Collection<User> = db_client.collection::<User>("users");
+    validate_object(&user_collection, input.user_id).await?;
+    validate_order_items(&db_client, &input.order_items).await?;
+    Ok(())
+}
+
+/// Checks if all order item parameters are the system (MongoDB database populated with events).
+///
+/// Used before creating orders.
+async fn validate_order_items(
+    db_client: &Database,
+    order_items: &BTreeSet<OrderItemInput>,
+) -> Result<()> {
+    let product_variant_version_collection: mongodb::Collection<ProductVariantVersion> =
+        db_client.collection::<ProductVariantVersion>("product_variant_versions");
+    let product_item_collection: mongodb::Collection<ProductItem> =
+        db_client.collection::<ProductItem>("product_items");
+    let tax_rate_version_collection: mongodb::Collection<TaxRateVersion> =
+        db_client.collection::<TaxRateVersion>("tax_rate_versions");
+    let shipment_method_collection: mongodb::Collection<ShipmentMethod> =
+        db_client.collection::<ShipmentMethod>("shipment_methods");
+    let product_variant_version_ids = order_items
```

Comment:
Producs and ProductVariants can be made non-publicly visible
we probably need a new event in catalog which notifies you of that, right?
because otherwise a user could just remember or guess a UUID, which is probably bad...

Reply:
If you agree, please open an issue on catalog, and I will build that feature today

Reply:
Alternatively, we could introduce a sync request

Reply:
Opened the issue and will add event endpoints to update the local copies accordingly

Reply:
Filters non-publicly-visible product variants. Products are not synced by events here